### PR TITLE
fix: remain

### LINKED ISSUE #24: remaining
## Phase 1: EmptyState Migration - Remaining Screens
1. **Part of:** https://github.com/rawadhossain/talawa-admin/issues/21
2. **PR Number:** 6 of 7
3. **Estimated Effort:** 3-4 hours
4. This is a detailed guide for the tasks to perform. There may be mistakes, so you'll need to use your discretion in tackling the various code changes and PRs. Use your initiative to have the best outcome for the long term maintainability of the code base.

## Overview
Complete the migration for all remaining screens with empty state patterns (Funds, Pledges, Leaderboard, OrganizationPeople).

## Files to Modify
* `src/screens/FundCampaignPledge/FundCampaignPledge.tsx`
* `src/screens/UserPortal/Pledges/Pledges.tsx`
* `src/screens/Leaderboard/Leaderboard.tsx`
* `src/screens/OrganizationPeople/OrganizationPeople.tsx`
* `src/screens/BlockUser/BlockUser.tsx`
* Corresponding test files

### FundCampaignPledge.tsx
* [ ]  Replace Stack/inline empty state with EmptyState
* [ ]  Add appropriate icon
* [ ]  Tests updated

### Pledges.tsx
* [ ]  Replace Stack-based empty state with EmptyState
* [ ]  Add pledges icon
* [ ]  Tests updated

### Leaderboard.tsx
* [ ]  Replace "noVolunteers" empty state with EmptyState
* [ ]  Add leaderboard/trophy icon
* [ ]  Tests updated

### OrganizationPeople.tsx
* [ ]  Replace empty state with EmptyState
* [ ]  Add people/groups icon
* [ ]  Tests updated

### BlockUser.tsx
* [ ]  Replace "no users found" / "no spammer found" with EmptyState
* [ ]  Handle context-specific messages
* [ ]  Tests updated

## Icon Suggestions
* FundCampaignPledge: `volunteer_activism` or `handshake`
* Pledges: `assignment` or `task_alt`
* Leaderboard: `emoji_events` or `leaderboard`
* OrganizationPeople: `groups` or `people`
* BlockUser: `block` or `person_off`

## Sample Implementation
```tsx
<EmptyState
  icon="emoji_events"
  message="noVolunteersYet"
  description="volunteersWillAppearHere"
  dataTestId="leaderboard-empty-state"
/>
```
## Links
* **Comprehensive Plan:** https://github.com/rawadhossain/talawa-admin/issues/21#issuecomment-3689596882
* **Parent Issue:** https://github.com/rawadhossain/talawa-admin/issues/21

### PR DESCRIPTION
**Pull request by rawadhossain**
**What kind of change does this PR introduce?**
Refactor -- migration for all remaining screens with empty state patterns (Funds, Pledges, Leaderboard, OrganizationPeople).

**Issue Number:**
Fixes #24 

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

* **New Features**
  * Standardized EmptyState across screens for consistent no-data UX
  * Leaderboard: localized medal labels/columns and added Weekly/Monthly timeframes
  * Pledges: localized "+N more" indicator and toggle for pledged/raised

* **Internationalization**
  * Added translations for new UI keys (leaderboard, pledges, common, user search) across EN/ES/FR/HI/ZH

* **Style**
  * Refined data-grid, avatar, progress-bar and global utility classes

* **Tests**
  * Added/expanded assertions for empty-state rendering, localized headers, and empty searches

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/screens/BlockUser/BlockUser.spec.tsx [hunk 1]
```diff
@@ -289,6 +289,9 @@ describe('BlockUser Component', () => {
 
       // Should show empty state
       await waitFor(() => {
+        expect(
+          screen.getByTestId('block-user-empty-state'),
+        ).toBeInTheDocument();
         expect(screen.getByText(/noUsersFound/i)).toBeInTheDocument();
       });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/BlockUser/BlockUser.spec.tsx:296]: current setup is fine and intentional since we migrated to emptystate migration

File: src/screens/OrganizationPeople/OrganizationPeople.tsx [hunk 2]
```diff
@@ -512,9 +490,11 @@ function OrganizationPeople(): JSX.Element {
...
       noRowsOverlay: () => (
-        <Stack height="100%" alignItems="center" justifyContent="center">
-          {tCommon('notFound')}
-        </Stack>
+        <EmptyState
+          icon="groups"
+          message={tCommon('notFound')}
+          dataTestId="organization-people-empty-state"
+        />
       ),
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/OrganizationPeople/OrganizationPeople.tsx:498]: We intentionally use tCommon('notFound') here to keep the empty state generic across tabs (members/admins/users). This matches existing usage patterns and avoids over-specializing the message. No change needed.

File: src/style/app-fixed.module.css [hunk 3]
```diff
@@ -119,6 +119,7 @@
 
   --font-size-header: 16px;
   --font-size-table-body: 14px;
+  --avatar-placeholder-bg: #ccc;
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/style/app-fixed.module.css:124]: The avatar size already has a safe fallback where it’s consumed, so adding a root-level --avatar-size isn’t necessary at this point. We’ll keep it minimal for now.

File: public/locales/hi/translation.json [hunk 4]
```diff
@@ -1734,6 +1740,7 @@
   "common": {
     "signOut": "साइन आउट",
     "signingOut": "साइन आउट हो रहा है...",
-    "retryPrompt": "सेशन रद्द करने में विफल। पुनः प्रयास करें?"
+    "retryPrompt": "सेशन रद्द करने में विफल। पुनः प्रयास करें?",
+    "breadcrumb": "ब्रेडक्रम्ब"
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply public/locales/hi/translation.json]: addressed in commit 26e5ac5d8ff8be7d3f2d01b1209f272e0165e0e9

File: src/screens/Leaderboard/Leaderboard.tsx [hunk 5]
```diff
@@ -168,19 +156,31 @@ function leaderboard(): JSX.Element {
...
             <>
-              <img src={bronze} alt="bronze" className={styles.rankings} />
+              <img
+                src={bronze}
+                alt={t('bronzeMedal')}
+                className={styles.rankings}
+              />
             </>
           );
         } else return <>{params.row.rank}</>;
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/Leaderboard/Leaderboard.tsx]: not necessary, currently not causing any harm

File: public/locales/en/translation.json [hunk 6]
```diff
@@ -9,7 +9,10 @@
     "weekly": "This Week",
     "monthly": "This Month",
     "yearly": "This Year",
-    "noVolunteers": "No Volunteers Found!"
+    "noVolunteers": "No Volunteers Found!",
+    "goldMedal": "gold",
+    "silverMedal": "silver",
+    "bronzeMedal": "bronze"
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply public/locales/en/translation.json]: Addressed in commit e23ac5da003e8302d7d27f2f712f453b5b6d250b

File: src/screens/Leaderboard/Leaderboard.tsx [hunk 7]
```diff
@@ -315,12 +312,22 @@ function leaderboard(): JSX.Element {
...
-        sx={dataGridStyle}
+        className={`${styles.dataGridNoHover} ${styles.dataGridRounded}`}
+        sx={{
+          '&.MuiDataGrid-root .MuiDataGrid-cell:focus-within': {
+            outline: 'none !important',
+          },
+          '&.MuiDataGrid-root .MuiDataGrid-columnHeader:focus-within': {
+            outline: 'none',
+          },
+        }}
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/Leaderboard/Leaderboard.tsx]: @coderabbitai should i make changes in `Pledges.tsx` and `FundCampaignPledge.tsx` as well? like this:
Pledges.tsx file:
```tsx
sx={{
          '&.MuiDataGrid-root .MuiDataGrid-cell:focus-within': {
            outline: '2px solid #1778f2',
            outlineOffset: '-2px',
          },
          '&.MuiDataGrid-root .MuiDataGrid-columnHeader:focus-within': {
            outline: '2px solid #1778f2',
            outlineOffset: '-2px',
          },
        }}
```

FundCampaignPledge.tsx file:
```tsx
sx={{
          '&.MuiDataGrid-root .MuiDataGrid-cell:focus-within': {
            outline: '2px solid #1778f2',
            outlineOffset: '-2px',
          },
          '&.MuiDataGrid-root .MuiDataGrid-columnHeader:focus-within': {
            outline: '2px solid #1778f2',
            outlineOffset: '-2px',
          },
        }}
```

[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/Leaderboard/Leaderboard.tsx]: addressed in latest commit 4aa2ebe7588c07622df357ed8679e1b3349f4c8c
outline: 'none' — removed focus indicators and Replaced all hex color codes (#1778f2) with CSS variables (var(--primary-theme-color))
No hex colors found in any of the three files (grep confirmed)
CSS variables properly used: var(--primary-theme-color)
No linter errors
Focus indicators still functional for accessibility

File: src/screens/FundCampaignPledge/FundCampaignPledge.tsx [hunk 8]
```diff
@@ -24,25 +24,13 @@ import type {
...
-  '&.MuiDataGrid-root .MuiDataGrid-columnHeader:focus-within': {
-    outline: 'none',
-  },
-  '& .MuiDataGrid-row:hover': { backgroundColor: 'transparent' },
-  '& .MuiDataGrid-row.Mui-hovered': { backgroundColor: 'transparent' },
-  '& .MuiDataGrid-root': { borderRadius: '0.5rem' },
-  '& .MuiDataGrid-main': { borderRadius: '0.5rem' },
-};
-
 const fundCampaignPledge = (): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/FundCampaignPledge/FundCampaignPledge.tsx:34]: Totally unnecessary, current setup is perfectly fine and working as expected. Keeping as it is now

**Comment by rawadhossain** [comment]
@coderabbitai full review and reevaluate all resolved items and include the following points:

1. **Test Coverage & Quality Validation:**
   
   1. Validate whether any updated test files:
      
      1. Cover all use cases including edge cases, error scenarios, and success paths
      2. Have valid, properly structured tests following Jest and React Testing Library best practices
      3. Provide 100% test code coverage for all modified/added functions, components, and branches
      4. Include proper mocking of GraphQL queries/mutations, external dependencies, and API calls
      5. Test both user interactions and state changes comprehensively
      6. Require any additional tests for uncovered scenarios
   2. For each modified file with logic or UI, verify corresponding test files exist and are updated
   3. Check that test files follow naming conventions (*.test.tsx, *.test.ts, *.spec.tsx, *.spec.ts)
   4. Validate that tests don't include disabled tests (it.skip, describe.skip) unless explicitly documented
   5. Ensure mock data cleanup is performed (as required by pre-commit hooks)
2. **Code Quality & Repository Standards:**
   
   1. Verify TypeScript compliance:
      
      1. No use of `any` type without explicit justification
      2. Proper interface/type definitions for all props, state, and function parameters
      3. No TypeScript errors or warnings
   2. Validate GraphQL implementation:
      
      1. Queries/mutations are properly typed and follow existing patterns in src/GraphQl/
      2. No duplicate or redundant queries (check for pagination inconsistencies)
      3. Error handling is implemented for all GraphQL operations
   3. Check React component best practices:
      
      1. Functional components with proper hooks usage
      2. Material-UI component patterns are followed consistently
      3. No prop drilling; proper use of context when needed
      4. Proper cleanup in useEffect hooks
   4. Verify internationalization (i18n):
      
      1. All user-facing strings use i18next translation keys
      2. New translation keys are added to locale files
      3. No hardcoded English text in UI components
   5. Validate accessibility:
      
      1. Proper ARIA labels and roles
      2. Keyboard navigation support
      3. Screen reader compatibility
3. **Documentation & Code Comments:**
   
   1. Verify auto-generated documentation is updated (as enforced by Check-AutoDocs job)
   2. Check for adequate JSDoc comments on public functions and complex logic
   3. Ensure README.md or related docs are updated if new features are added
   4. Validate that Table of Contents is updated (as required by pre-commit hooks)
4. **Identify Issues:**
   
   1. Lines of code that are missing tests (list specific line numbers)
   2. Any unnecessary files that have been submitted (build artifacts, node_modules, .env files, IDE configs)
   3. Unused imports or dead code (as checked by knip)
   4. Console.log statements or debug code left in production files
   5. Files that should be excluded from the PR scope
5. **PR Requirements Validation:**
   
   1. Verify the PR meets the requirements of the linked issue:
      
      1. All acceptance criteria are addressed
      2. No out-of-scope changes are included
   2. Validate compliance with all previously suggested improvements and comments:
      
      1. All review comments have been addressed
      2. Requested changes have been implemented correctly
   3. Check PR metadata:
      
      1. Proper title and description
      2. Linked to appropriate issue(s)
      3. Target branch is correct (typically `develop`)
   4. Verify no sensitive information is exposed (as checked by Check-Sensitive-Files job)
6. **CI/CD Pipeline Compatibility:**
   
   1. Validate that code changes won't break any of the 17+ CI jobs
   2. Ensure pre-commit hook requirements are met:
      
      1. Code formatting (Prettier)
      2. Linting (ESLint)
      3. Type checking (TypeScript)
   3. Check that changes don't introduce performance regressions for the 12-shard test execution
7. **Security & Best Practices:**
   
   1. No security vulnerabilities introduced
   2. No exposed API keys, tokens, or credentials
   3. Proper input validation and sanitization
   4. No SQL injection or XSS vulnerabilities in any data handling
8. **Implementation Correctness:**
   
   * For the specific implementation file(s) being tested:
     
     * Verify the implementation logic is correct and free of bugs
     * Validate that queries target the correct database tables/entities
     * Check that foreign key relationships are correctly enforced
     * Confirm business logic constraints are properly implemented
     * Any bugs, logic errors, or incorrect implementations discovered in the file(s) under test MUST be fixed in this PR before approval
9. **Review Decision:**
   
   1. When critical issues are found, clearly state that changes are required in your response but DO NOT append  the APPROVED tag.
   2. Only append  the APPROVED tag when ALL criteria are satisfied and zero changes are required:
      
      1. Complete test coverage (≥95%)
      2. No code quality issues, security concerns, or bugs
      3. No bugs or logic errors in the implementation file(s) under test
      4. Full implementation of issue requirements
      5. Compliance with all previous feedback
      6. No unnecessary files
10. **Focus Areas:**

* Prioritize substantive issues over minor stylistic suggestions
* Focus on issues that would cause CI pipeline failures or production bugs
* Highlight any deviations from established codebase patterns

**Note:** Test-focused PRs should maintain clear scope boundaries. However, this does NOT exempt bugs in the specific file(s) being tested - those must be fixed in the same PR. Only bugs in unrelated files may be deferred to separate issues.

File: public/locales/es/translation.json [hunk 9]
```diff
@@ -9,7 +9,14 @@
...
     "yearly": "Este Año",
-    "noVolunteers": "¡No Se Encontraron Voluntarios!"
+    "noVolunteers": "¡No Se Encontraron Voluntarios!",
+    "goldMedal": "oro",
+    "silverMedal": "plata",
+    "bronzeMedal": "bronce",
+    "rank": "Rango",
+    "volunteer": "Voluntario",
+    "email": "Correo electrónico",
+    "hoursVolunteered": "Horas Voluntarias"
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply public/locales/es/translation.json:19]: no need, current setup is fine

File: src/screens/Leaderboard/Leaderboard.tsx [hunk 10]
```diff
@@ -315,12 +312,24 @@ function leaderboard(): JSX.Element {
...
           noRowsOverlay: () => (
-            <Stack height="100%" alignItems="center" justifyContent="center">
-              {t('noVolunteers')}
-            </Stack>
+            <EmptyState
+              icon="emoji_events"
+              message={t('noVolunteers')}
+              dataTestId="leaderboard-empty-state"
+            />
           ),
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/Leaderboard/Leaderboard.tsx]: The empty state behavior is already covered by the existing tests, and search filtering logic is exercised separately. Adding an additional search-empty test would be redundant at this stage, so we’ll keep the current coverage.

File: public/locales/fr/translation.json [hunk 11]
```diff
@@ -854,6 +865,7 @@
     "searchByFirstName": "Recherche par prénom",
     "searchByLastName": "Rechercher par nom de famille",
     "noSpammerFound": "Aucun spammeur trouvé",
+    "noUsersFound": "Aucun utilisateur trouvé",
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply public/locales/fr/translation.json:878]: currently it is fine, no unnecesary change needed

File: src/style/app-fixed.module.css [hunk 12]
```diff
@@ -10057,3 +10060,78 @@ form {
...
+
+.progressBarHeight {
+  height: 1.5rem;
+  font-size: 0.9rem;
+}
+
+.memberNameFontSize {
+  font-size: 15px;
+}
+
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/style/app-fixed.module.css:10314]: unnecessary change, not needed. currently it is working just fine

File: public/locales/fr/translation.json [hunk 13]
```diff
@@ -9,7 +9,14 @@
...
     "yearly": "Cette Année",
-    "noVolunteers": "Aucun Bénévole Trouvé!"
+    "noVolunteers": "Aucun Bénévole Trouvé!",
+    "goldMedal": "or",
+    "silverMedal": "argent",
+    "bronzeMedal": "bronze",
+    "rank": "Rang",
+    "volunteer": "Bénévole",
+    "email": "E-mail",
+    "hoursVolunteered": "Heures Bénévoles"
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply public/locales/fr/translation.json:19]: not needed, current text is perfectly understandable. Keeping as it is

**Comment by rawadhossain** [comment]
@coderabbitai full review based on this prompt:
https://github.com/rawadhossain/talawa-admin/pull/25#issuecomment-3691239199

**Comment by rawadhossain** [comment]
@coderabbitai write me a summary on the pr what i did to resolve the issue in precise, short few bullets
